### PR TITLE
BW-1256 /runs/{run id} endpoint_KP

### PR DESCRIPTION
**Pull request by kpierre13**

### CODE REVIEW

File: engine/src/main/scala/cromwell/webservice/routes/wes/CromwellMetadata.scala [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+  import CromwellMetadata._
+
+  def wesRunLog: WesRunLog = {
+    val workflowParams = submittedFiles.inputs.map(JsonParser(_).asJsObject)
+    val workflowTags = submittedFiles.labels.map(JsonParser(_).asJsObject)
+    val workflowEngineParams = submittedFiles.options.map(JsonParser(_).asJsObject)
+
+    val workflowRequest = WesRunRequest(workflow_params = workflowParams,
+      workflow_type = submittedFiles.workflowType.getOrElse("Unable to find workflow type"),
+      workflow_type_version = submittedFiles.workflowTypeVersion.getOrElse("Unable to find workflow version"),
```
Comment (cjllanwarne) [inline engine/src/main/scala/cromwell/webservice/routes/wes/CromwellMetadata.scala:40]: If a user doesn't submit these values, I think it's strange for the metadata to respond with "unable to find". I'd probably just leave them as empty strings (or maybe "not supplied"?)... or does Cromwell normally fill them in even if the request didn't include them?

File: engine/src/main/scala/cromwell/webservice/routes/wes/WesRunRoutes.scala [hunk 2]
```diff
@@ -20,42 +27,49 @@ trait WesRunRoutes {
...
-            get {
-              parameters(("page_size".as[Int].?, "page_token".?)) { (pageSize, pageToken) =>
-                completeCromwellResponse(listRuns(pageSize, pageToken, serviceRegistryActor))
-              }
+          get {
+            parameters(("page_size".as[Int].?, "page_token".?)) { (pageSize, pageToken) =>
+              WesRunRoutes.completeCromwellResponse(WesRunRoutes.listRuns(pageSize, pageToken, serviceRegistryActor))
             }
           }
+          concat(
```
Comment (cjllanwarne) [inline engine/src/main/scala/cromwell/webservice/routes/wes/WesRunRoutes.scala:37]: We're already inside a `concat`... do we still need this nested `concat`?

File: engine/src/main/scala/cromwell/webservice/routes/wes/WesRunRoutes.scala [hunk 3]
```diff
@@ -20,42 +27,49 @@ trait WesRunRoutes {
 
   lazy val runRoutes: Route =
     pathPrefix("ga4gh" / "wes" / "v1") {
+      concat(
         pathPrefix("runs") {
-          pathEnd {
```
Comment (cjllanwarne) [inline engine/src/main/scala/cromwell/webservice/routes/wes/WesRunRoutes.scala:24]: Do you not still need the `pathEnd` to indicate "if there's nothing else in the URL, do this"?

File: engine/src/main/scala/cromwell/webservice/routes/wes/WesRunRoutes.scala [hunk 4]
```diff
@@ -20,42 +27,49 @@ trait WesRunRoutes {
...
+    // FIXME: How to handle next_page_token in response?
+    metadataQueryRequest(Seq.empty[(String, String)], serviceRegistryActor).map(RunListResponse.fromMetadataQueryResponse)
   }
-}
 
-def listRuns(pageSize: Option[Int], pageToken: Option[String], serviceRegistryActor: ActorRef): Future[WesResponse] = {
-  // FIXME: to handle - page_size, page_token
-  // FIXME: How to handle next_page_token in response?
-  metadataQueryRequest(Seq.empty[(String, String)], serviceRegistryActor).map(RunListResponse.fromMetadataQueryResponse)
+  def runLog(workflowId: String, request: WorkflowId => BuildMetadataJsonAction, serviceRegistryActor: ActorRef): Future[WesResponse] = {
```
Comment (cjllanwarne) [inline engine/src/main/scala/cromwell/webservice/routes/wes/WesRunRoutes.scala:67]: OOI, why is the `request` a function? Couldn't you just take in the workflow ID here and build the `GetSingleWorkflowMetadataAction` directly within this function?

File: engine/src/main/scala/cromwell/webservice/routes/wes/CromwellMetadata.scala (hunk 1 above)
Comment (kpierre13) [reply engine/src/main/scala/cromwell/webservice/routes/wes/CromwellMetadata.scala:40]: I decided to change this to "not supplied," as the POST endpoint allows for these values to be input, and I think it makes it more obvious to a user what is going on here.

File: engine/src/main/scala/cromwell/webservice/routes/wes/WesRunRoutes.scala (hunk 3 above)
Comment (kpierre13) [reply engine/src/main/scala/cromwell/webservice/routes/wes/WesRunRoutes.scala:24]: I'm curious as to how this slipped in here. On my current branch, this `pathEnd` is not present.

File: engine/src/main/scala/cromwell/webservice/routes/wes/WesRunRoutes.scala (hunk 4 above)
Comment (kpierre13) [reply engine/src/main/scala/cromwell/webservice/routes/wes/WesRunRoutes.scala:67]: Investigating...

Comment (aednichols) [reply engine/src/main/scala/cromwell/webservice/routes/wes/WesRunRoutes.scala:65]: I don't have a good answer for why, but we copied this from existing code at my suggestion

https://github.com/broadinstitute/cromwell/blob/develop/engine/src/main/scala/cromwell/webservice/routes/MetadataRouteSupport.scala#L151

**Review by cjllanwarne — APPROVED** [review]

**Review by aednichols — APPROVED** [review]

**Pull request merged**